store/store.go

```diff
@@ -141,19 +141,19 @@ func (s *DefaultStore) LoadCommit(height uint64) (*types.Commit, error) {
 func (s *DefaultStore) LoadCommitByHash(hash [32]byte) (*types.Commit, error) {
 	commitData, err := s.db.Get(getCommitKey(hash))
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error retrieving commit from hash %v: %w", hash, err)
 	}
 	commit := new(types.Commit)
 	err = commit.UnmarshalBinary(commitData)
-	return commit, err
+	return commit, fmt.Errorf("error decoding binary data of Commit into object: %w", err)
```

Comment:
Gotcha! In original case, if error was `nil`, `nil` was returned. Now, you're wrapping everytime, even if error is `nil`.
```suggestion
	if err != nil {
		return nil, fmt.Errorf("error decoding binary data of Commit into object: %w", err)
	}
	return commit, nil
```

Reply:
Updated 👍